generate_engagement_analysis_files.py

```diff
@@ -0,0 +1,103 @@
+
+Logger.set_project_name("IMAQAL")
+log = Logger(__name__)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Computes engagement data and stores it in multiple CSV sheets for"
+                                                 "analysis and visualization")
+
+    parser.add_argument("user", help="User launching this program")
+    parser.add_argument("pipeline_configuration_file_path", metavar="pipeline-configuration-file",
+                        help="Path to the pipeline configuration json file"),
+    parser.add_argument("demog_map_json_input_dir", metavar="demog-map-json-input-dir",
+                        help="Path to a directory to read per episode demog map .json files. "
+                             "The files should be in the format <rqa_raw_field>_demog_map.json")
+    parser.add_argument("engagement_csv_output_dir", metavar="engagement-csv-output-dir",
+                        help="Directory to write engagement CSV files to")
+
+
+    args = parser.parse_args()
+
+    user = args.user
+    pipeline_configuration_file_path = args.pipeline_configuration_file_path
+    demog_map_json_input_dir = args.demog_map_json_input_dir
+    engagement_csv_output_dir = args.engagement_csv_output_dir
+
+    # Load the pipeline configuration file
+    log.info("Loading Pipeline Configuration File...")
+    with open(pipeline_configuration_file_path) as f:
+        pipeline_configuration = PipelineConfiguration.from_configuration_file(f)
+
+    # Infer which RQA coding plans to use from the pipeline name.
+    if pipeline_configuration.pipeline_name == "q4_pipeline":
+        log.info("Running Q4 pipeline")
+        PipelineConfiguration.RQA_CODING_PLANS = PipelineConfiguration.Q4_RQA_CODING_PLANS
+    elif pipeline_configuration.pipeline_name == "q5_pipeline":
+        log.info("Running Q5 pipeline")
+        PipelineConfiguration.RQA_CODING_PLANS = PipelineConfiguration.Q5_RQA_CODING_PLANS
+    elif pipeline_configuration.pipeline_name == "q6_pipeline":
+        log.info("Running Q6 pipeline")
+        PipelineConfiguration.RQA_CODING_PLANS = PipelineConfiguration.Q6_RQA_CODING_PLANS
+    else:
+        assert pipeline_configuration.pipeline_name == "full_pipeline", "PipelineName must be either 'a quartely pipeline name' or 'full pipeline'"
+        log.info("Running full Pipeline")
+        PipelineConfiguration.RQA_CODING_PLANS = PipelineConfiguration.FULL_PIPELINE_RQA_CODING_PLANS
+
+    # Create a list of rqa_raw_fields
+    rqa_raw_fields = []
+    for plan in PipelineConfiguration.RQA_CODING_PLANS:
+        rqa_raw_fields.append(plan.raw_field)
+
+    log.info(f'Computing unique, lifetime-active and per-show participants ...' )
+    engagement_map = {}  # of uid -> name of shows participated in and their demographics data
+    uuids = set()  # unique uids that participated in the entire project
+    lifetime_active_uuids = []  # uids that participated in each radio show
```

Comment:
I don't understand how these can be lifetime active for anything other than a fill pipeline run

Reply:
This stores the uids of individuals who participated each week which we then get the len() of the list to get the total lifetime active individuals. By reading each show demog map we can capture each uid that participated in that show as though its a full pipeline run.

Reply:
This has the wrong variable name then.

Consider `active_uuds_per_week`

Reply:
renamed to `lifetime_activations_per_show` @lukechurch PTAL